tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -1605,24 +1604,182 @@ def _test_memory_based_scheduling_enabled_false(
     assert_that(slurm_commands.get_node_attribute("queue1-st-ondemand1-i1-1", "Memory")).is_equal_to("3891")
     assert_that(slurm_commands.get_node_attribute("queue1-dy-ondemand1-i3-1", "Memory")).is_equal_to("31129")
 
-    # TODO: Add functional tests for memory-based scheduling
+    # Upload files for memory allocation tests
+    remote_command_executor._copy_additional_files(
+        [
+            str(test_datadir / "memory_allocation_chars.c"),
```

Comment:
Cool that you have implemented a C program to allocate a given amount of memory.
Just out of my curiosity: why not simply have a bash command allocating an array?

Reply:
I guess I'm more familiar with memory allocation in C rather than bash :)

Reply:
Fair enough :-)